### PR TITLE
Upgrade to rails 6

### LINKED ISSUE #2153: upgrade to Rails 6

### LINKED ISSUE #2566: Upgrade to rails 6

### PR DESCRIPTION
**Pull request by Br3nda**
closes #2566 
closes #2153

### CODE REVIEW

**Comment by Br3nda** [comment]
Brakeman is complaining about not using `attr_accessible` but this was removed from rails in rails 6 -- so it seems Brakeman is not up to date with rails 6

File: .rubocop.yml [hunk 1]
```diff
@@ -60,3 +60,75 @@ Rails/SkipsModelValidations:
...
+Lint/RaiseException:
+  Enabled: true
+Lint/StructNewOverride:
+  Enabled: true
+Style/AccessorGrouping:
+  Enabled: true
+Style/ArrayCoercion:
+  Enabled: true
+Style/BisectedAttrAccessor:
+  Enabled: true
```
Comment (Br3nda) [inline .rubocop.yml]:
```suggestion
Style/BisectedAttrAccessor:
  Enabled: false
```

File: bin/yarn [hunk 2]
```diff
@@ -1,6 +1,4 @@
 #!/usr/bin/env ruby
```
Comment (Br3nda) [inline bin/yarn]:
```suggestion
#!/usr/bin/env ruby
 # frozen_string_literal: true 
```

File: config/puma.rb [hunk 3]
```diff
@@ -1,45 +1,38 @@
-# frozen_string_literal: true
-
-# Overview of config:
-# https://devcenter.heroku.com/articles/deploying-rails-applications-with-the-puma-web-server
-
 # Puma can serve each request in a thread from an internal thread pool.
```
Comment (Br3nda) [inline config/puma.rb:3]:
```suggestion
# frozen_string_literal: true

# Puma can serve each request in a thread from an internal thread pool.
```

File: bin/setup [hunk 4]
```diff
@@ -1,8 +1,5 @@
 #!/usr/bin/env ruby
-# frozen_string_literal: true
-
 require 'fileutils'
```
Comment (Br3nda) [inline bin/setup:4]:
```suggestion
# frozen_string_literal: true

require 'fileutils'
```

**Comment by cesy** [comment]
Thanks so much for doing this upgrade.

File: app/views/places/_map_attribution.html.haml [hunk 5]
```diff
@@ -3,4 +3,3 @@ Map data &copy;
 contributors under
 = link_to "ODbL", "https://www.openstreetmap.org/copyright"
 | Imagery &copy;
```
Comment (Br3nda) [inline app/views/places/_map_attribution.html.haml]:
```suggestion
```

File: app/views/places/_map_attribution.html.haml [hunk 6]
```diff
@@ -2,5 +2,3 @@ Map data &copy;
 = link_to "OpenStreetMap", "https://openstreetmap.org"
 contributors under
 = link_to "ODbL", "https://www.openstreetmap.org/copyright"
-| Imagery &copy;
-= link_to "CloudMade", "https://cloudmade.com"
```
Comment (Br3nda) [inline app/views/places/_map_attribution.html.haml:6]: this company used to produce the leaflet widget, but that staff member moved elsewhere in like 2012 and took the code with them and made it open source. the credit is no longer needed

**Comment by Br3nda** [comment]
@cesy just a heads up, i no longer have a job as a Rails dev - so i'm still keep up with this stuff for fun, but eventaully my knowledge of Rails is going to fade out with new rails version.

**Comment by Br3nda** [comment]
it's been two weeks so i'm just gonna merge this one now

**Pull request merged**